builtin/staker/linkedlist/linked_list.go

```diff
@@ -0,0 +1,189 @@
+	if err := l.prev.Set(address, oldTail, false); err != nil {
+		return err
+	}
+
+	// Update tail pointer
+	l.tail.Set(&address, false)
+
+	return l.count.Add(big.NewInt(1))
+}
+
+// Remove extracts an address from anywhere in the list, reconnecting adjacent nodes and clearing removed node's pointers
+func (l *LinkedList) Remove(address thor.Address) error {
+	if address.IsZero() {
+		return nil
+	}
+
+	prev, err := l.prev.Get(address)
+	if err != nil {
+		return err
+	}
+
+	next, err := l.next.Get(address)
+	if err != nil {
+		return err
+	}
+
+	// If address is not in the list (no prev and not head)
+	if prev.IsZero() && !l.isHead(address) {
+		return nil // not in list
+	}
+
+	// Update previous node's next pointer
+	if !prev.IsZero() {
+		if err := l.next.Set(prev, next, false); err != nil {
+			return err
+		}
+	} else {
+		// This is the head, update head pointer
+		l.head.Set(&next, false)
+	}
+
+	// Update next node's prev pointer
+	if !next.IsZero() {
+		if err := l.prev.Set(next, prev, false); err != nil {
+			return err
+		}
+	} else {
+		// This is the tail, update tail pointer
+		l.tail.Set(&prev, false)
+	}
+
+	// Clear the removed node's pointers
+	if err = l.next.Set(address, thor.Address{}, false); err != nil {
+		return err
+	}
+	if err = l.prev.Set(address, thor.Address{}, false); err != nil {
+		return err
+	}
+
+	return l.count.Sub(big.NewInt(1))
+}
+
+// Pop removes and returns the oldest entry (head) for FIFO processing order
+func (l *LinkedList) Pop() (thor.Address, error) {
+	head, err := l.head.Get()
+	if err != nil {
+		return thor.Address{}, errors.New("no head present")
+	}
+
+	if head.IsZero() {
+		return thor.Address{}, errors.New("list is empty")
+	}
+
+	// otherwise, remove and return
+	if err := l.Remove(head); err != nil {
+		return thor.Address{}, err
+	}
+	return head, nil
+}
+
+// Peek returns the next address to be processed without removing it from the queue
+func (l *LinkedList) Peek() (thor.Address, error) {
+	return l.head.Get()
+}
+
+// Len returns the current number of addresses in the staker queue
+func (l *LinkedList) Len() (*big.Int, error) {
+	return l.count.Get()
+}
+
+// Iter traverses the list in FIFO order, calling callback for each address until completion or error
+func (l *LinkedList) Iter(callback func(thor.Address) error) error {
+	ptr, err := l.head.Get()
+	if err != nil {
+		return err
+	}
+
+	for !ptr.IsZero() {
+		if err := callback(ptr); err != nil {
+			return err
+		}
+
+		next, err := l.next.Get(ptr)
+		if err != nil {
+			return err
+		}
+
+		if next.IsZero() {
+			break
+		}
+		ptr = next
+	}
+
+	return nil
+}
+
+// Next returns the successor address in the list, or zero address if at the end
+func (l *LinkedList) Next(address thor.Address) (thor.Address, error) {
+	return l.next.Get(address)
+}
+
+// isHead checks if the given address is the head of the list
+func (l *LinkedList) isHead(address thor.Address) bool {
+	head, err := l.head.Get()
+	if err != nil {
+		return false
```

Comment:
Small one: do we wanna still return the error to let the caller know what was wrong? I'm not sure but the caller might decide to do different things based on `false, nil` instead of `false, Error("...")`